src/coreclr/jit/compiler.cpp

```diff
@@ -5730,8 +5748,18 @@ void Compiler::compCompileFinish()
         unsigned profCallCount = 0;
         if (opts.jitFlags->IsSet(JitFlags::JIT_FLAG_BBOPT) && fgHaveProfileData())
         {
-            assert(fgBlockCounts[0].ILOffset == 0);
-            profCallCount = fgBlockCounts[0].ExecutionCount;
+            bool foundEntrypointBasicBlockCount = false;
+            for (UINT32 iSchema = 0; iSchema < fgPgoSchemaCount; iSchema++)
+            {
+                if ((fgPgoSchema[iSchema].InstrumentationKind == ICorJitInfo::PgoInstrumentationKind::BasicBlockIntCount) &&
+                    (fgPgoSchema[iSchema].ILOffset == 0))
```

Comment:
My suggestion instead would be to make a unique InstrumentationKind to describe that instead of using a magic ILOffset. Details like this are why there are 3 separate descriptive fields associated with each run of data collection details. This may require the parser in the jit to be able to handle both forms in the future, but that's just the reality of supporting a static pgo system.